src/v/cluster/metrics_reporter.cc

```diff
@@ -262,6 +270,10 @@ metrics_reporter::build_metrics_snapshot() {
         snapshot.id_hash = license->checksum;
     }
 
+    snapshot.license_violation
+      = feature_report.any()
+        && (!license.has_value() || license.value().is_expired());
+
```

Comment:
I think splitting this into two separate fields `has_enterprise_features` and `has_valid_license` would be useful because filtering for enterprise clusters on its own seems useful. Wdyt?

Reply:
ya fair point.